langlib/langlib-test/src/test/resources/test-src/regexp_test.bal

```diff
@@ -2588,6 +2588,96 @@ function testCharClassesWithMultipleRangesAndAtoms() returns error? {
     assertEquality("b", (<regexp:Span>res9).substring());
 }
 
+function testRegexpWithUnicodeChars() {
+    string startChar = string `\u{0d9c}`;
+    string endChar = string `\u{1F310}`;
+    regexp:Span? resA1 = regexp:find(re = re `\u{1F30D}`, str = "😀 Hello, 🌍! ❤️");
+    assertTrue(resA1 is regexp:Span);
+    regexp:Span? resA2 = regexp:find(re `\u{1F30D}`, "🌍😀 Hello! ❤️", 5);
+    assertTrue(resA2 is ());
+    regexp:Span? resA3 = regexp:find(re `\u{1F30D}`, "😀 Hello 🌍! ❤️", 6);
+    assertTrue(resA3 is regexp:Span);
+    regexp:Span? resA4 = regexp:find(re `\u{1F31F}`, "😀 Hello, 🌍! ❤️");
+    assertTrue(resA4 is ());
+    regexp:Span? resA5 = regexp:find(re `\u{0d9c}`, "පරිගණක 10");
+    assertTrue(resA5 is regexp:Span);
+    regexp:Span? resA6 = regexp:find(re `[${startChar}${endChar}]`, "😀 Hello, 🌏! ❤️");
+    assertTrue(resA6 is regexp:Span);
+    regexp:Span? resA7 = regexp:find(re `[${startChar}${endChar}]`, "😀 Hello,! ❤️");
+    assertTrue(resA7 is regexp:Span);
+    // Invalid code point range
+    regexp:Span? resA8 = regexp:find(re `\u{FF3AD}`, "😀 Hello, 🌍! ❤️");
+    assertTrue(resA8 is ());
+
+    regexp:Span[] resB1 = regexp:findAll(re `\u{1F30D}`, "😀 Hello, 🌍! ❤️");
+    assertTrue(resB1.length() == 1);
+    regexp:Span[] resB2 = regexp:findAll(re `\u{1F30D}`, "🌍😀 Hello! ❤️", 5);
+    assertTrue(resB2.length() == 0);
+    regexp:Span[] resB3 = regexp:findAll(re `\u{1F30D}`, "😀 Hello 🌍! 🌍❤️", 6);
+    assertTrue(resB3.length() == 2);
+    regexp:Span[] resB4 = regexp:findAll(re `\u{1F31F}`, "😀 Hello, 🌍! ❤️");
+    assertTrue(resB4.length() == 0);
+    regexp:Span[] resB5 = regexp:findAll(re `\u{0d9c}`, "පරිගණක 10");
+    assertTrue(resB5.length() == 1);
+    regexp:Span[] resB6 = regexp:findAll(re `[${startChar}${endChar}]`, "😀 Hello, 🌏! ❤️");
+    assertTrue(resB6.length() == 1);
+    regexp:Span[] resB7 = regexp:findAll(re `[${startChar}${endChar}]`, "😀 Hello,! ❤️");
+    assertTrue(resB7.length() == 1);
+    // Invalid code point range
+    regexp:Span[] resB8 = regexp:findAll(re `\u{FF3AD}`, "😀 Hello, 🌍! ❤️");
+    assertTrue(resB8.length() == 0);
+
+    regexp:Groups? resC1 = regexp:findGroups(re `o\u{1F30D}`, "😀 Hello🌍! ❤️o🌍");
+    assertTrue(resC1 is regexp:Groups);
+    regexp:Groups? resC2 = regexp:findGroups(re `l\u{1F30D}`, "😀 Hello🌍! ❤️o🌍");
+    assertTrue(resC2 is ());
+
+    regexp:Groups[] resD1 = regexp:findAllGroups(re `o\u{1F30D}`, "😀 Hello🌍! ❤️o🌍");
+    assertTrue(resD1.length() == 2);
```

Comment:
https://github.com/ballerina-platform/ballerina-lang/pull/40368/files#r1197345321

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/40368/commits/d10a307642318c20ec015c816bc8541bb2e5a572.